libraries/tools/kotlin-gradle-plugin-integration-tests/src/test/kotlin/org/jetbrains/kotlin/gradle/KotlinGradlePluginIT.kt

```diff
@@ -1053,4 +1053,39 @@ class KotlinGradleIT : BaseGradleIT() {
             assertTasksExecuted(":lib1:compileDebugKotlin")
         }
     }
+
+    /** Regression test for KT-38692. */
+    @Test
+    fun testIncrementalWhenNoKotlinSources() = with(
+        Project("kotlinProject")
+    ) {
+        setupWorkingDir()
+        assertTrue(this.allKotlinFiles.toList().isNotEmpty())
+        build(":compileKotlin") {
+            assertSuccessful()
+            assertTasksExecuted(":compileKotlin")
+        }
+
+        // Remove all Kotlin sources and also modify buildscript classpath
+        allKotlinFiles.forEach { assertTrue(it.delete()) }
+        projectDir.resolve("src/main/java/Sample.java").also {
```

Comment:
If we don't add java source here, the compile kotlin task will skip?

Reply:
Correct, and in that case Gradle removes outputs. I'd like to test scenario in which task runs, but there are no Kotlin sources.

Reply:
nit: maybe add that comment as it's not clear from the test?